src/util/edit.cc

```diff
@@ -214,6 +250,40 @@ data::BinData EditEngine::getDataFromEditNode(const EditNode& edit_node,
   return edit_node.fragment_->data(edit_node.offset_ + offset, size);
 }
 
+void EditEngine::remap(size_t pos, size_t old_size, size_t new_size) {
+  // This variable can be overflowed for `old_size` > `new_size`,
+  // but it will be OK when we add this difference to some address.
+  size_t offset = new_size - old_size;
+
+  auto pos_next_it = address_mapping_.upperBound(pos);
+  assert(pos_next_it != address_mapping_.cbegin());
+
+  QMap<size_t, EditNode> new_address_mapping;
+  for (auto it = address_mapping_.cbegin(); it != pos_next_it; ++it) {
+    new_address_mapping.insert(it.key(), it.value());
+  }
+
+  size_t insert_pos = pos + old_size;
```

Comment:
Could you add an explanation in a comment why do you pick this certain value? (`pos + old_size`)

Reply:
Done